addons/web/static/tests/views/calendar_tests.js

```diff
@@ -2107,6 +2107,67 @@ QUnit.module('Views', {
         calendar.destroy();
     });
 
+    QUnit.test('timzeone does not affect calendar with date field', function (assert) {
+        assert.expect(8);
+
+        var calendar = createView({
+            View: CalendarView,
+            model: 'event',
+            data: this.data,
+            arch:
+            '<calendar date_start="start_date" mode="month">'+
+                '<field name="name"/>'+
+                '<field name="start_date"/>'+
+            '</calendar>',
+            archs: archs,
+            viewOptions: {
+                initialDate: initialDate,
+            },
+            mockRPC: function (route, args) {
+                if (args.method === "create") {
+                    assert.strictEqual(args.args[0].start_date, "2016-12-20 00:00:00");
+                }
+                if (args.method === "write") {
+                    assert.step(args.args[1].start_date);
+                }
+                return this._super(route, args);
+            },
+            session: {
+                getTZOffset: function () {
+                    return -2400; // 40 hours timezone
```

Comment:
It doesn't represent reality...

Reply:
changed to 120 (2 hours) so the test fail with `[day before expected days] 22:00:00` instead of with the fix: `[expected day] 00:00:00`